src/TodoList.js

```diff
@@ -1,41 +1,67 @@
 class TodoList {
-  constructor () {
+  constructor() {
     this.id = 0
     this.items = []
   }
 
-  create (str) {
+  create(str) {
     this.id++
-    const item = { id: this.id, text: str, status: 'incomplete' }
+    const item = {
+      id: this.id,
+      text: str,
+      status: 'incomplete',
+      creationDate: 'date that the ToDo was created'
+    }
     this.items.push(item)
     return item
   }
 
-  showAll () {
+  showAll() {
     return this.items
   }
 
-  setComplete (id) {
+  setComplete(id) {
     const item = this.findBy(id)
     item.status = 'complete'
     return item
   }
 
-  getByStatus (status) {
-    return this.items.filter(item => item.status === status)
+  getByStatus(status) {
+    return this.items.filter((item) => item.status === status)
   }
 
-  findBy (id) {
-    const item = this.items.find(item => item.id === id)
+  findBy(id) {
+    const item = this.items.find((item) => item.id === id)
     if (item === undefined) throw new Error('Item not found')
     return item
   }
 
-  deleteBy (id) {
+  deleteBy(id) {
     const item = this.findBy(id)
     const index = this.items.indexOf(item)
     return this.items.splice(index, 1)[0]
   }
+
+  showMaxChar() {
+    // eslint-disable-next-line no-unused-vars
+    const maxText = this.items.map((item) => {
+      if (item.text.length > 20) {
+        return { ...item, text: `${item.text.substring(0, 20)}....` }
+      }
+      return { maxText }
+    })
+    return maxText
+  }
```

Comment:
This is very close, you just need to make sure to return the item inside the map instead of maxText:

```js
showMaxChar() {
    // eslint-disable-next-line no-unused-vars
    const maxText = this.items.map((item) => {
      if (item.text.length > 20) {
        return { ...item, text: `${item.text.substring(0, 20)}....` }
      }
      return item
    })
    return maxText
  }
```